### ISSUE TITLE
Talk to DAC about possible solutions for vendor issues

### ISSUE THREAD

**Issue by owenatgov** [alphagov/govuk-design-system#4194]
## What
Set up a discussion with DAC to talk about our solutions for the vendor issues:

- https://github.com/alphagov/govuk-frontend/issues/5310 where the solution was https://github.com/alphagov/govuk-frontend/pull/5089
- https://github.com/alphagov/govuk-frontend/issues/5309 where the solution was https://github.com/alphagov/govuk-frontend/pull/5305
- https://github.com/alphagov/govuk-frontend/issues/5308 where the solution was https://github.com/alphagov/govuk-frontend/pull/5372

## Why
We want to get DAC's input on all 3 of these as they improve the experience for AT users but are imperfect. More detail can be found in the issues themselves.

The status if each issue currently is that:
- the details with VoiceOver on iOS Safari issue we consider fixed to the point of closing the original issue, however we'd liek to still get DAC's input as the solution doesn't work for older versions of iOS Safari. We believe this is a vendor level problem that is very difficult for us to consistently fix and test.
- we're confident in our solution for the details with Dragon issue but we want to verify it as it changes the announcement content for a lot of AT.
- our solution for file upload with Dragon is an improvement but still has issues to do with browser security policy. We want DAC's opinion on if this is worthwhile or not before proceeding.

## Who needs to work on this
Accessibility specialist

## Done when

- [x] Email was sent to DAC
- [x] DAC responded
- [x] Decisions are made on what to do with the issues

**Comment by selfthinker** [comment]
I have [drafted an email](https://docs.google.com/document/d/1iMp0LOKm8d9Vfszp_YdUB1jEaxpoJKI7ZHIMcE2CtJ8/edit?usp=sharing) I'd like to send DAC (can only be viewed internally). I plan to send it on Tuesday.

**Comment by selfthinker** [comment]
I've sent the email today.

**Comment by selfthinker** [comment]
DAC responded on Wednesday. I have added most of the email conversation to the same document I referenced above.

## Summary of the response

### Details with Dragon
They said adding ARIA roles etc is likely to cause more problems, so not worth doing. They also said Dragon users are "not currently adversely impacted" because most Dragon users are used to the Details components not working properly and would therefore automatically use the known workarounds.
They keep on mentioning the idea that we could use what we use for the accordion component (a button opening and closing things with JavaScript). But we wouldn't change the component like that as that is creating our custom version of an HTML element that exists natively.

### File upload with Dragon
They said our planned implementation makes sense. And that the majority of users wouldn't need to use the file upload more than once, so are not likely to notice that it won't work the second time.

### Details with VoiceOver
They hadn't responded to this issue, but I hadn't asked a question about that. I only informed them that we fixed it with some more details.

## Decisions
* Based on the above, I suggest to keep the Details component the way it is. ([Creating a browser extension to fix the issue](https://github.com/alphagov/govuk-frontend/issues/5448) is still an option, but not a priority.)
* We should proceed with our [plan to enhance the file upload component](https://github.com/alphagov/govuk-design-system/issues/4242).
